### PR TITLE
e2e File plugin test scenarios

### PR DESCRIPTION
**Pull request by sawantpritam**
@rmstar @itsankit-google Please review.

### CODE REVIEW

File: e2e-test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>io.cdap.plugin</groupId>
+    <artifactId>e2e-test</artifactId>
+    <version>2.9.0-SNAPSHOT</version>
```
Comment (rmstar) [inline e2e-test/pom.xml]: develop is 2.10.0-SNAPSHOT - https://github.com/cdapio/hydrator-plugins/blob/develop/pom.xml#L23

File: e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+    When Select plugin: "File" from the plugins list as: "Source"
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "BigQuery" from the plugins list as: "Sink"
+    Then Connect plugins: "File" and "BigQuery" to establish connection
+    Then Navigate to the properties page of plugin: "File"
+    Then Enter input plugin property: "referenceName" with value: "FileReferenceName"
+    Then Enter input plugin property: "path" with value: "outputFieldTestFile"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Enter File plugin property: override field name with value: "overrideField"
+    Then Select File plugin property: override field data type with value: "overrideDataType"
```
Comment (rmstar) [inline e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature]: Which field's data type is overridden?

File: e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+    Given Open Datafusion Project to configure pipeline
+    When Select plugin: "File" from the plugins list as: "Source"
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "BigQuery" from the plugins list as: "Sink"
+    Then Connect plugins: "File" and "BigQuery" to establish connection
+    Then Navigate to the properties page of plugin: "File"
+    Then Enter input plugin property: "referenceName" with value: "FileReferenceName"
+    Then Enter input plugin property: "path" with value: "blobFile"
+    Then Select dropdown plugin property: "format" with option value: "blob"
+    Then Enter input plugin property: "maximumSplitSize" with value: "fileMaxSplitSize"
```
Comment (rmstar) [inline e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature]: Do we end up with multiple splits with this split size (120) ?

File: e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+    Given Open Datafusion Project to configure pipeline
+    When Select plugin: "File" from the plugins list as: "Source"
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "BigQuery" from the plugins list as: "Sink"
+    Then Connect plugins: "File" and "BigQuery" to establish connection
+    Then Navigate to the properties page of plugin: "File"
+    Then Enter input plugin property: "referenceName" with value: "FileReferenceName"
+    Then Enter input plugin property: "path" with value: "readRecursivePath"
+    Then Select dropdown plugin property: "format" with option value: "csv"
+    Then Select radio button plugin property: "readFilesRecursively" with value: "true"
```
Comment (rmstar) [inline e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature]: Do we have subdirectories in our input to test this case?

File: e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+    Then Enter input plugin property: "table" with value: "bqTargetTable"
+    Then Click plugin property: "truncateTable"
+    Then Click plugin property: "updateTableSchema"
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Enter runtime argument value "csvFile" for key "fileSourcePath"
+    Then Enter runtime argument value "csvFormat" for key "fileSourceFormat"
+    Then Enter runtime argument value "skipHeaderTrue" for key "fileSourceSkipHeader"
```
Comment (rmstar) [inline e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature]: can we have a test case where skipHeader is set to false?

File: e2e-test/src/test/resources/pluginParameters.properties [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+## FILE-PLUGIN-PROPERTIES-START
+incorrectFilePath=invalidPath
+csvFile=testdata/file/CSV_TEST.csv
+csvFileSchema=[{"key":"EmployeeDepartment","value":"string"},{"key":"Employeename","value":"string"},\
+  {"key":"Salary","value":"int"},{"key":"workhours","value":"int"}]
+csvFileOutputSchema={ "type": "record", "name": "text", "fields": [ \
+  { "name": "EmployeeDepartment", "type": "string" }, { "name": "Employeename", "type": "string" }, \
+  { "name": "Salary", "type": "int" }, { "name": "workhours", "type": "int" } ] }
+csvAllDataTypeFile=testdata/file/CSV_DATATYPE_TEST_1.csv
+csvAllDataTypeFileSchema=[{"key":"id","value":"int"},{"key":"name","value":"string"},\
```
Comment (rmstar) [inline e2e-test/src/test/resources/pluginParameters.properties]: Make sure we test all supported data types. Looks like we're only testing int, string, boolean, double

File: e2e-test/pom.xml (hunk 1 above)
Comment (sawantpritam) [reply e2e-test/pom.xml]: Removed.

File: e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature (hunk 2 above)
Comment (sawantpritam) [reply e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature]: "overrideField" - id
from int to float

File: e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature (hunk 3 above)
Comment (sawantpritam) [reply e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature]: No. Seems to be open defect.

File: e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature (hunk 4 above)
Comment (sawantpritam) [reply e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature]: Yes.

File: e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature (hunk 5 above)
Comment (sawantpritam) [reply e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature]: Added scenario with skipHeader set to false

File: e2e-test/src/test/resources/pluginParameters.properties (hunk 6 above)
Comment (sawantpritam) [reply e2e-test/src/test/resources/pluginParameters.properties]: Updated.

**Pull request closed without merging**